src/Storages/ColumnsDescription.cpp

```diff
@@ -714,6 +718,12 @@ bool ColumnsDescription::hasCompressionCodec(const String & column_name) const
 
 CompressionCodecPtr ColumnsDescription::getCodecOrDefault(const String & column_name, CompressionCodecPtr default_codec) const
 {
+    if (column_name == BlockNumberColumn.name)
```

Comment:
It's slightly strange that we add this logic into general structure `ColumnsDescription`. We can do it, but I'd prefer to add assertion here (column_name != BlockNumberColumn.name) and add special method for BlockNumber column.

Reply:
addressed it